src/ucp/api/ucp.h

```diff
@@ -2413,6 +2504,20 @@ ucs_status_ptr_t ucp_ep_flush_nb(ucp_ep_h ep, unsigned flags,
 ucs_status_ptr_t ucp_ep_flush_nbx(ucp_ep_h ep, const ucp_request_param_t *param);
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief Query attributes of a specific endpoint.
+ *
+ * This routine fetches information about the endpoint.
+ * 
+ * @param [in]    ep   Endpoint to query.
+ * @param [inout] attr Filled with attributes of the endpoint.
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_ep_estimate_perf(ucp_ep_h ep, ucp_ep_perf_attr_t *attr);
```

Comment:
Why limit the query just for performance ?

Reply:
@shamisp pls see https://github.com/openucx/ucx/pull/6403#discussion_r586448976

Reply:
perf offline discussion with @shamisp, suggest to rename this API to ucp_ep_assess_perf.
@tonycurtis WDYT?

Reply:
My first thought was "eval" might be better, and that's already been suggested.

(The comment needs updating too if that's not a W.I.P. and you could put "estimate" there if you think it would be helpful)

Reply:
@shamisp are you ok with `ucp_ep_evaluate_perf`?

Reply:
👍

Reply:
@gleon99 can you pls rename (and docs, if needed) to `ucp_ep_evaluate_perf`?

Reply:
@yosefe , done